Feature/asset types

tnagorra opened the pull request:
## Depends on

- https://github.com/mapswipe/mapswipe-firebase/pull/34

## Changes
    
- add external_url on CommonAsset
    - add file and external_url to CommonAsset model
    - make file non-mandatory
    - add oneOf validation for file and external_url
- add input_type on ProjetAsset: AOI_GEOMETRY, OBJECT_IMAGE, COVER_IMAGE
    - add input_type on ProjectAsset filter
    - add export_type on ProjectAsset filter
- add asset_type_specifics on ProjectAsset
    - define ObjectImage and AoiGeometry pydantic models
    - add validation for AOI_GEOMETRY on serializer
    - calculate bbox, center and area for AOI_GEOMETRY on serializer
- clear older DEBUG related project assets
- add min_zoom and max_zoom for raster tiles
- add tutorial specifics for validate_image project type
- increase max tasks for validate image task using dataset
- convert id fields in object image asset to numeric string

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

tnagorra committed:
feat(asset)!: add external_url and input_type on project assets

- add external_url on CommonAsset
    - add file and external_url to CommonAsset model
    - make file non-mandatory
    - add oneOf validation for file and external_url
- add input_type on ProjetAsset: AOI_GEOMETRY, OBJECT_IMAGE, COVER_IMAGE
    - add input_type on ProjectAsset filter
    - add export_type on ProjectAsset filter
- add asset_type_specifics on ProjectAsset
    - define ObjectImage and AoiGeometry pydantic models
    - add validation for AOI_GEOMETRY on serializer
    - calculate bbox, center and area for AOI_GEOMETRY on serializer

tnagorra committed:
fix(asset)!: clear older DEBUG related project assets

frozenhelium committed:
feat(validate-image): update project specifics

tnagorra committed:
fix(firebase): use feature id instead of index for validate task

tnagorra committed:
refactor(validate): move feature grouping logic to utils

tnagorra committed:
feat(validate-image): implement processing of validate image

frozenhelium committed:
feat(validate-image): add specifics for tutorial

frozenhelium committed:
feat(validate-image): increase max image asset count for object image

frozenhelium committed:
refactor(raster-tile): add min and max zoom

tnagorra committed:
fix(validate-image): convert ids from coco to numeric string

tnagorra committed:
feat(asset): remove mimetype from asset mutation

- add validation for all input types for project asset creation
- add input type and validation in tutorial asset creation
- remove mimetype based filtering for project asset retrieval

tnagorra committed:
feat(validate-image)!: handle generation of tutorial data on firebase

- change id in validate image to numeric string

tnagorra committed:
fix(asset)!: rename objectImage to object_image

tnagorra committed:
feat(firebase): use absolute uri while using asset in firebase

thenav56 commented on apps/common/utils.py:
```diff
@@ -0,0 +1,23 @@
+
+
+def get_absolute_uri(file: files.FieldFile | None) -> str | None:
+    if not file:
```
also need to add file.name check

tnagorra committed:
fix(file): add is_file_empty to check for file with empty content

- add typing for file content on asset serializers
- update firebase submodule to develop
